ext/soap/php_encoding.c

```diff
@@ -384,17 +384,17 @@ static xmlNodePtr master_to_xml_int(encodePtr encode, zval *data, int style, xml
 	    Z_OBJCE_P(data) == soap_var_class_entry) {
 		zval *ztype, *zdata, *zns, *zstype, *zname, *znamens;
 		encodePtr enc = NULL;
-		HashTable *ht = Z_OBJPROP_P(data);
+		zval rv;
 
-		if ((ztype = zend_hash_str_find_deref(ht, "enc_type", sizeof("enc_type")-1)) == NULL ||
-		    Z_TYPE_P(ztype) != IS_LONG) {
+		ztype = zend_read_property(soap_var_class_entry, Z_OBJ_P(data), "enc_type", sizeof("enc_type")-1, 1, &rv);
```

Comment:
I don't know whether it's better to explicitly specify the class entry or I should just use `Z_OBJCE_P(data)` in all the `zend_read_property()` calls. Also, currently I pass `1` to the `silent` parameter, but probably it shouldn't be the case?

Reply:
As these are public properties, it doesn't really make a difference. But generally, this is the scope from which you access the property, so explicitly passing soap_var_class_entry is semantically correct.

Note that read_property doesn't return NULL, an undefined property is `&EG(uninitialized_zval)`. I think using silent mode and explicitly reporting an error is okay in this file, because it uses some super broken error handling -- passing through the normal exception would be ideal, but it's probably not easily possible.

I suspect that you also need to add a ZVAL_DEREF here, don't think read_property does that by itself.

Reply:
The fact that `&EG(uninitialized_zval)` is returned means that the type is effectively null. So if you perform some type check, you can just drop the NULL pointer check entirely.

It may make sense to add a read_property_deref helper as this is probably going to be a common pattern for this type of code.

Reply:
Thanks for the hints! I like the idea of adding the helper.  And I'll try to fix these issues the following days

Reply:
My newest commit (https://github.com/php/php-src/pull/6759/commits/07d76460872514e650ed52bf92430bbd8da3325a) addresses the review comments, so now only one test fails. Although error handling of property reading is still broken. :(